lib/core/test/unit/Cardano/Wallet/Primitive/TypesSpec.hs

```diff
@@ -413,6 +419,187 @@ spec = do
+                    let withinBounds = time >= timeMinimum
+                    checkCoverage $
+                        cover 10      withinBounds  "within bounds" $
+                        cover 10 (not withinBounds) "out of bounds" $
+                        case epochFloor sps time of
+                            Nothing -> not withinBounds
+                            Just en -> time >= epochStartTime sps en
+
+        it "epochStartTime (epochSucc (epochFloor t)) > t" $
+            withMaxSuccess 1000 $ property $
+                \(SlotParametersAndTimePoint sps time) -> do
+                    let timeMinimum = epochStartTime sps minBound
+                    let withinBounds = time >= timeMinimum
+                    checkCoverage $
+                        cover 10      withinBounds  "within bounds" $
+                        cover 10 (not withinBounds) "out of bounds" $
+                        case epochFloor sps time of
+                            Nothing -> not withinBounds
+                            Just e1 -> case epochSucc e1 of
+                                Nothing -> e1 == maxBound
+                                Just e2 -> time < epochStartTime sps e2
+
+    describe "Epoch arithmetic: epochCeiling: boundary conditions" $ do
+
+        it "epochCeiling . epochStartTime == id" $
+            withMaxSuccess 1000 $ property $ \(sps, epoch) ->
+                epochCeiling sps (epochStartTime sps epoch)
+                    === Just epoch
+
+        it "epochCeiling . utcTimePred . epochStartTime == id" $
+            withMaxSuccess 1000 $ property $ \(sps, epoch) ->
+                epoch > minBound ==> do
+                    let fun = epochCeiling sps
+                            . utcTimePred
+                            . epochStartTime sps
+                    Just epoch === fun epoch
+
+        it "epochPred . epochCeiling . utcTimeSucc . epochStartTime == id" $
+            withMaxSuccess 1000 $ property $ \(sps, epoch) ->
+                epoch < maxBound ==> do
+                    let fun = (epochPred =<<)
+                            . epochCeiling sps
+                            . utcTimeSucc
+                            . epochStartTime sps
+                    Just epoch === fun epoch
+
+        it "epochCeiling (epochStartTime minBound) == minBound" $
+            withMaxSuccess 1000 $ property $ \sps ->
+                epochCeiling sps (epochStartTime sps minBound)
+                    === Just minBound
+
+        it "epochCeiling (utcTimePred (epochStartTime minBound)) == minBound" $
+            withMaxSuccess 1000 $ property $ \sps ->
+                epochCeiling sps (utcTimePred (epochStartTime sps minBound))
+                    === Just minBound
+
+        it "epochCeiling (epochStartTime maxBound) == maxBound" $
+            withMaxSuccess 1000 $ property $ \sps ->
+                epochCeiling sps (epochStartTime sps maxBound)
+                    === Just maxBound
```

Comment:
**[remark]**

the same remark as above to `"epochCeiling (epochStartTime minBound) == minBound"`

Reply:
See: https://github.com/input-output-hk/cardano-wallet/pull/1114#discussion_r354095995